src/base/api.lua

```diff
@@ -658,7 +658,7 @@
 		-- If a tool was not found, check to see if there is a function in the
 		-- table to check against.  For each function in the table, check if
 		-- the value is allowed (break early if so).
-		if not canonical then
+		if not canonical and type(field.allowed) == "table" then
```

Comment:
I think this is the correct change, but I'm wondering if we should output a warning if it's not the expected type?